### PR TITLE
import app data

### PR DESCRIPTION
**Pull request by brianolson**
Import Applications/Stateful-TEAL data and make it available through the API

### CODE REVIEW

**Comment by brianolson** [comment]
Ping. This should be ready to go. I have some extra testing infrastructure I may add later.

File: importer/importer.go [hunk 1]
```diff
@@ -82,15 +83,26 @@ func (imp *dbImporter) ImportDecodedBlock(blockContainer *types.EncodedBlockCert
...
 			assetid = uint64(stxn.Txn.ConfigAsset)
+			if assetid == 0 {
+				assetid = block.TxnCounter - uint64(len(block.Payset)) + uint64(intra) + 1
+			}
 		case 4:
 			assetid = uint64(stxn.Txn.XferAsset)
 		case 5:
 			assetid = uint64(stxn.Txn.FreezeAsset)
+		case 6:
+			assetid = uint64(stxn.Txn.ApplicationID)
```
Comment (winder) [inline importer/importer.go:102]: Probably good to rename this `createableId`, or whatever the general term is

File: idb/postgres_migrations.go [hunk 2]
```diff
@@ -122,6 +139,7 @@ func init() {
 	migrations = []migrationFunc{
 		m0fixupTxid,
 		m1fixupBlockTime,
+		m2apps,
```
Comment (winder) [inline idb/postgres_migrations.go:142]: This worked out really nicely.

File: types/types.go [hunk 3]
```diff
@@ -202,8 +202,40 @@ type (
 		SenderRewards   MicroAlgos `codec:"rs"`
 		ReceiverRewards MicroAlgos `codec:"rr"`
 		CloseRewards    MicroAlgos `codec:"rc"`
+		EvalDelta       EvalDelta  `codec:"dt"`
 	}
 
+	// EvalDelta stores StateDeltas for an application's global key/value store, as
+	// well as StateDeltas for some number of accounts holding local state for that
+	// application
+	EvalDelta struct {
```
Comment (winder) [inline types/types.go:211]: I just did a little digging and see that this is part of ApplyData, the delta portion makes a lot more sense now.

**Pull request merged**